java-support/src/main/java/io/cloudstate/javasupport/CloudState.java

```diff
@@ -222,7 +232,77 @@ public CloudState registerCrdtEntity(
       Descriptors.FileDescriptor... additionalDescriptors) {
     services.put(
         descriptor.getFullName(),
-        new CrdtStatefulService(factory, descriptor, newAnySupport(additionalDescriptors)));
+        system ->
+            new CrdtStatefulService(factory, descriptor, newAnySupport(additionalDescriptors)));
+
+    return this;
+  }
+
+  /**
+   * Register an annotated Controller service.
+   *
+   * <p>The controller class must be annotated with {@link
+   * io.cloudstate.javasupport.controller.Controller}.
+   *
+   * @param controller The controller object.
+   * @param descriptor The descriptor for the service that this controller implements.
+   * @param additionalDescriptors Any additional descriptors that should be used to look up protobuf
+   *     types when needed.
+   * @return This Cloudstate builder.
+   */
+  public CloudState registerController(
```

Comment:
So according to the controllerspec.proto one can register 'custom' controllers for message routing? I like that a lot as it treats the controller/router/thing on the same level as a user function. Is that how its intended to work?